packages/realm/src/GeoSpatial.ts

```diff
@@ -0,0 +1,164 @@
+  | {
+      latitude: number;
+      longitude: number;
+      altitude?: number;
+    }
+  | CanonicalGeoPoint
+  | GeoPosition;
+
+/**
+ * Represents a circle in spherical geometry that can be used for the definition of geospatial queries.
+ */
+export type GeoCircle = {
+  /** The center of the circle. */
+  center: GeoPoint;
+  /** The radius of the circle in radians. You can use {@link kmToRadians} and {@link miToRadians}
+   * to respectively convert kilometers and miles to radians. */
+  distance: number;
+};
+
+/**
+ * Represents a polygon in spherical geometry that can be used for the definition of geospatial queries.
+ * The polygon is comprised of at least one outer ring and optionally multiple internal rings representing holes with the following restrictions:
+ * - Each ring must contains at least 3 distinct points, where the first and the last point must be the same to indicate a closed ring (this means that each ring
+ * must have at least 4 points).
+ * - The interior rings must be entirely inside the outer ring.
+ * - Rings can share vertices but not edges.
+ * - No ring may be empty.
+ */
+export type GeoPolygon =
+  | {
+      outerRing: GeoPoint[];
+      holes?: GeoPoint[][];
+    }
+  | CanonicalGeoPolygon;
+
+/**
+ * Represents a box in spherical geometry that can be used for the definition of geospatial queries.
+ * This is a short-hand for the equivalent {@link GeoPolygon}.
+ */
+export type GeoBox = {
+  /** The bottom left point of the box. */
+  bottomLeft: GeoPoint;
+  /** The top right point of the box. */
+  topRight: GeoPoint;
+};
+
+/** @internal */
+export function circleToBindingGeospatial(circle: GeoCircle): binding.Geospatial {
+  return binding.Geospatial.makeFromCircle({
+    center: toBindingGeoPoint(circle.center),
+    radiusRadians: circle.distance,
+  });
+}
+
+/** @internal */
+export function boxToBindingGeospatial(box: GeoBox): binding.Geospatial {
+  return binding.Geospatial.makeFromBox({
+    lo: toBindingGeoPoint(box.bottomLeft),
+    hi: toBindingGeoPoint(box.topRight),
+  });
+}
+
+/** @internal */
+export function polygonToBindingGeospatial(polygon: GeoPolygon): binding.Geospatial {
+  let points: binding.GeoPoint_Relaxed[][];
+  if ("type" in polygon) {
+    points = toBindingGeoPointArray(polygon.coordinates);
+  } else {
+    points = toBindingGeoPointArray([polygon.outerRing].concat(polygon.holes ?? []));
+  }
+
+  return binding.Geospatial.makeFromPolygon({
+    points,
+  });
+}
+
+function toBindingGeoPoint(p: GeoPoint): binding.GeoPoint_Relaxed {
+  if (Array.isArray(p)) {
+    return { longitude: p[0], latitude: p[1], altitude: p[2] };
+  } else if ("type" in p) {
+    return { longitude: p.coordinates[0], latitude: p.coordinates[1], altitude: p.coordinates[2] };
+  } else {
+    return p;
+  }
+}
+
+function toBindingGeoPointArray(arr: GeoPoint[][]): binding.GeoPoint_Relaxed[][] {
+  return arr.map((ring) => ring.map((p) => toBindingGeoPoint(p)));
+}
+
+const earthRadiusKm = 6378.1;
```

Comment:
Is it common to use International Astronomical Union's value and not the one provided by International Union of Geodesy and Geophysics?

Reply:
Maybe 👀